lib/spack/spack/cmd/find.py

```diff
@@ -140,9 +140,15 @@ def find(parser, args):
 
     # Exit early if no package matches the constraint
     if not query_specs and args.constraint:
-        msg = "No package matches the query: {0}"
-        msg = msg.format(' '.join(args.constraint))
-        tty.msg(msg)
+        msg = "No installed package"
+        if args.constraint:
+            msg += " matches the query: {0}".format(' '.join(args.constraint))
+        msg2 = "If you are looking for installable packages try `spack list"
+        if args.constraint:
+            msg2 += " {0}`".format(' '.join(args.constraint))
+        else:
+            msg2 += "`"
+        tty.msg(msg, msg2)
```

Comment:
Below you use `tty.msg` twice, here you use it once with two arguments. I haven't run this yet, but I feel like `tty.msg` twice gives two `==>` prompts whereas `tty.msg` with two arguments only gives one `==>` prompt. Am I right? I don't care which we use, but we should be consistent.

Reply:
In `list.py`. You have one message for "available packages (matching ...)" and one for "try spack find".

Reply:
Ah yeah, but between these two are the packages

Reply:
So we really can't put them on one

Reply:
Exactly, so maybe we should split this `tty.msg` with two arguments into two calls to `tty.msg` so that the format is the same?

Reply:
@tgamblin suggested this syntax, I really don't care either way